ingest/indexer/service/indexer_streaming_service.go

```diff
@@ -340,10 +345,15 @@ func (s *indexerStreamingService) ListenCommit(ctx context.Context, res abci.Res
 	s.blockUpdatesProcessUtils.SetChangeSet(changeSet)
 
 	// Create block processor
-	blockProcessor := blockprocessor.NewBlockProcessor(s.blockProcessStrategyManager, s.client, s.poolExtractor, s.keepers, s.blockUpdatesProcessUtils)
+	// Note the returned block processor can be either full or incremental depending on the strategy
+	// When node is syncing, it will be a full block processor
+	// When node is already synced, it will be an incremental block processor
+	blockProcessor := blockprocessor.NewBlockProcessor(s.blockProcessStrategyManager, s.client, s.poolExtractor, s.keepers, s.nodeStatusChecker, s.blockUpdatesProcessUtils)
 
 	// Process block.
 	if err := blockProcessor.ProcessBlock(sdkCtx); err != nil {
+		// Mark the error observed, so that the next block will reprocess the entire block.
+		s.blockProcessStrategyManager.MarkErrorObserved()
```

Comment:
I understand having this for syncing detecting but we should be careful because for the indexer we do not want to reprocess all state on post-start-up errors due to guaranteed delivery.

We might want to adjust the logic so that it is specific to synching detection but is not affected by errors after the first block.

This is needed for SQS because it does not have a persistent state.

Reply:
Hey Roman, when I tested it, using such call `MarkErrorObserved` looked to achieve what we want, i.e. no publishing until the sync is complete, but I may not be aware of other side effect. I will double check it and adjust the BlockProcessStrategyManager as needed. Thanks!